src/Models/AzureUtility.ts

```diff
@@ -431,17 +431,16 @@ export class AzureUtility {
       return undefined;
     }
 
-    const telemetryContext: TelemetryContext = {
-      properties: {
-        result: 'Succeeded',
-        error: '',
-        errorMessage: '',
-        subscription: subscription.description
-      },
-      measurements: {duration: 0}
-    };
+    const telemetryWorker = TelemetryWorker.getInstance(AzureUtility._context);
+    const telemetryContext = telemetryWorker.createContext();
+    telemetryContext.properties.subscription = subscription.description;
 
-    TelemetryWorker.sendEvent(EventNames.selectSubscription, telemetryContext);
+    try {

```

Comment:
If we could put try...catch in TelemetryWorker.sendEvent so that we don't need to explicitly try...catch when calliing sendEvent?

Reply:
I will fix this part of job in another standalone PR focus on error handling.